docs/user-guide/diff-strategies.md

```diff
@@ -0,0 +1,90 @@
+# Diff Strategies
+
+Argo CD calculates the diff between the desired state and the live
+state in order to define if an Application is out-of-sync. This same
+logic is also used in Argo CD UI to display the differences between
+live and desired states for all resources belonging to an application.
+
+Argo CD currently has 3 different strategies to calculate diffs:
+
+- **Legacy**: This is the main diff strategy used by default. It
+  applies a 3-way diff based on live state, desired state and
+  last-applied-configuration (annotation).
+- **Structured-Merge Diff**: Strategy automatically applied when
+  enabling Server-Side Apply sync option. 
+- **Server-Side Diff**: New strategy that invokes a Server-Side Apply
+  in dryrun mode delegating to Kube-API the action of calculating the
+  diff.
+
+## Structured-Merge Diff
+*Current Status: [Beta][1] (Since v2.5.0)*
```

Comment:
Is all of server-side apply considered beta? If not, I'm not sure we should retroactively mark this portion of that feature as beta.

Reply:
I think the way I see it is that Server-Side Apply is one thing and diff is something else. Server-Side apply works and I think we can even mark it as stable. The diff logic used automatically when SSA is enabled was never documented up to this point and that is what I willing to do with this doc. Structured-Merge Diff was never considered stable.